lisa/wa.py

```diff
@@ -17,14 +17,17 @@
 from collections.abc import Mapping
 from collections import defaultdict
 from operator import attrgetter
+from contextlib import closing
```

Comment:
I see we already import the whole of contextlib below, so I'll remove this and access it with a path